### DISCUSSION TITLE
"ESP-CSE Checksum failure" errors, has anyone seen them? How to 'fix' them? Should I worry?

### DISCUSSION THREAD

**Discussion by DurvalMenezes** [arendst/Tasmota#18083] [Q&A]
I have a pair of Sonoff POW R2 devices, and they both intermittently (at different times, but with similar frequency, from about half a dozen to a few dozen times a day) log this message: 

```
ESP-CSE: Checksum failure
```

I have googled for it, and specifically checked the tasmota github (both the issues and here in the discussion area) and also r/tasmota on Reddit, but haven't found much clarity.

Checking the sources, I find (in Tasmota current master, commit fbbf8ff) that this is logged at ./tasmota/tasmota_xnrg_energy/xnrg_02_cse7766.ino/CseSerialInput(); reading the code, it seems that this message is logged when a complete packet is received from the energy-measuring sensor *and* the checksum doesn't match && the packet header can't be 'synced' to its proper position (ie, there's no packet header marker '0x5A' anywhere in the packet).

So, here are my questions:
1) Has anyone seen these?
2) Should I be worried? 
3) How do I 'fix' that (ie, cure whatever is causing these messages)? 

TIA!

**Comment by barbudor** [comment]
Cause could be multiples
- bad contact on a chip (CSE or ESP)
- CPU too busy and loosing bytes

If it's only 10 times a day while readings occur every seconds or few seconds, I wouldn't call it a problem.

**Reply by sfromis** [reply]
Potential causes of bit errors includes nearby electrical noise.

**Comment by DurvalMenezes** [comment]
Thanks to everyone who responded. I'm gonna add some counters to Tasmota code to try and see how many errors vs successful reads are actually happening  and then decide on how better to handle this.
